runtime/Makefile

```diff
@@ -671,6 +693,53 @@ $(foreach platform,$(DOTNET_PLATFORMS),$(foreach rid,$(DOTNET_$(platform)_RUNTIM
 $(foreach platform,$(DOTNET_PLATFORMS),$(foreach rid,$(DOTNET_$(platform)_RUNTIME_IDENTIFIERS),$(foreach arch,$(DOTNET_$(rid)_ARCHITECTURES),$(eval $(call DotNetLibXamarinTemplate,$(DOTNET_$(rid)_SDK_PLATFORM),$(rid),$(arch),_CORECLR,-dotnet-coreclr,_CORECLR)))))
 $(foreach platform,$(DOTNET_PLATFORMS),$(foreach rid,$(DOTNET_$(platform)_RUNTIME_IDENTIFIERS),$(foreach arch,$(DOTNET_$(rid)_ARCHITECTURES),$(eval $(call DotNetLibXamarinTemplate,$(DOTNET_$(rid)_SDK_PLATFORM),$(rid),$(arch),_CORECLR_DEBUG,-dotnet-coreclr-debug,_CORECLR)))))
 
+#
+# DotNetLibExtensionTemplate builds libapp.a and lib[tv]extension.a
+#
+
+define DotNetLibExtensionTemplate
+
+# TODO note that we build libextension and libtvextension
+# for all cases, even though we need at most one of them.
+# for example, the output directory for tvos contains
+# libextension-dotnet.a (which is for iOS)
+# and libtvextension-dotnet.a (which is for tvOS)
+# it seems we should only need to build one.
```

Comment:
We select which extension library to build for which platform here:

https://github.com/xamarin/xamarin-macios/blob/427405c53cf9f23360992bbda1bd660214f7b4ff/runtime/Makefile#L570-L573

so we should only build the ones we want (and not libtvosextension-dotnet.a for iOS for instance).

If I look for `libtvextension-dotnet*` after a build, I only get results for tvOS:

```
$ find . -name 'libtvextension-dotnet*'
./.libs/tvsimulator/libtvextension-dotnet.arm64.a
./.libs/tvsimulator/libtvextension-dotnet.x86_64.a
./.libs/tvos/libtvextension-dotnet.arm64.a
```

maybe you had leftover files from previous attempts/builds?

Reply:
> maybe you had leftover files from previous attempts/builds

Possibly. I see what you mean now about how it should only be building what's needed.